spell/spell-reviewer-mainnet-checklist.md

```diff
@@ -1,369 +1,372 @@
-* [ ] `addresses_mainnet.sol` matches spell code
+* IF external contracts calls present (Not SubDAOs, e.g. Starknet)
+  * [ ] Target Contract don't block spell execution
+  * [ ] External call is NOT delegate call
+  * [ ] Target Contract doesn't have permissions on the Vat
+  * [ ] Target Contract doesn't do anything untoward (e.g. interacting with unsafe contracts)
+  * [ ] Contracts deployed via `CREATE2` (e.g. if it looks like a vanity address) do not have `selfdestruct` in their code
+  * [ ] MCD Pause Proxy doesn't give any approvals
+  * [ ] All possible actions of the Target Contract are documented
+  * [ ] Target contract is not upgradable
+  * [ ] Target Contract is included in the ChainLog
+  * [ ] Test Coverage is comprehensive
+* IF spell interacts with ChainLog
+  * [ ] ChainLog version is incremented based on update type
+    * Major -> New Vat (++.0.0)
+    * Minor -> Core Module (DSS) Update (e.g. Flapper) (0.++.0)
+    * Patch -> Collateral addition or addition/modification (0.0.++)
+  * [ ] New addresses are added to the `addresses_mainnet.sol`
+  * [ ] Changes are tested via `testNewOrUpdatedChainlogValues`
 * [ ] Ensure every spell variable is declared as `public`/`internal`
-* [ ] Ensure `immutable` visibility is only used when fetching addresses from the `ChainLog` via `DssExecLib.getChangelogAddress` and `constant` is used instead for static addresses
+* [ ] Ensure `immutable` visibility is only used when fetching addresses from the `ChainLog` via `DssExecLib.getChangelogAddress(key)` and `constant` is used instead for static addresses
   * [ ] Fetch addresses as type `address` and wrap with `Like` suffix interfaces inline (when making calls) unless archive patterns permit otherwise (Such as `MKR`)
   * [ ] Use the [DssExecLib Core Address Helpers](https://github.com/makerdao/dss-exec-lib/blob/master/src/DssExecLib.sol#L166) where possible (e.g. `DssExecLib.vat()`)
   * [ ] Where addresses are fetched from the `ChainLog`, the variable name must match the value of the ChainLog key for that address (e.g. `MCD_VAT` rather than `vat`), except where the archive pattern differs from this pattern (e.g. MKR)
 * [ ] Spell actions match the corresponding Exec Doc
-* [ ] Tests
+* Tests
   * [ ] Ensure each spell action has sufficient test coverage
     _List actions for which coverage was checked here_
   * [ ] Ensure every test function is declared as public if enabled or private if disabled
   * [ ] Ensure that the `DssExecLib.address` file is not being modified by the spell PR
   * [ ] Check all CI tests are passing as at the latest commit
     _Insert most recent commit hash where CI was passing_
   * [ ] Check all tests are passing locally using `make test`
-    * [ ] Ensure that only `ETH_RPC_URL` is being used from env (i.e. no `match`, `block` or similar are active in your env)
+  * [ ] Ensure that only `ETH_RPC_URL` is being used from env (i.e. no `match`, `block` or similar are active in your env)
```

Comment:
```suggestion
  * [ ] Ensure that only `ETH_RPC_URL` is being used from env (i.e. no `match`, `block` or similar are active in your env)
  * [ ] Check all tests are passing locally using `make test`
```

Reply:
At least the way I check it is to 1) first run `make test` 2) check first line of the output to match:
```
./scripts/test-dssspell-forge.sh no-match="" match="" block=""
```
That's why `make test` is above. 

But maybe then it's better to reformulate the env check to be:
> Ensure that `make test` output displays every optional parameter unset (ie first output line of `make test` is `./scripts/test-dssspell-forge.sh no-match="" match="" block=""`)

Does this sounds better now? Or how do you test it? Via `printenv | grep`?

Reply:
Actually, I don't think this is applicable.
There is env var for either match or block numbers in the latest versions of [Foundry](https://book.getfoundry.sh/reference/cli/forge/test#forge-test).

> > Ensure that make test output displays every optional parameter is unset (ie first output line of make test is `./scripts/test-dssspell-forge.sh no-match="" match="" block=""`)

Maybe this is good for explicitness. However, inspecting the env seems kind of pointless.

Regarding the order, it's just that you must set `ETH_RPC_URL` **before** running `make test`, that's why I suggested to swap those lines.

Reply:
Hmm... OMG, Foundry docs is a mess 😅 
I agree with using `printenv | grep "FOUNDRY_\|DAPP_"` then.